components/pkg-export-kubernetes/src/main.rs

```diff
@@ -0,0 +1,167 @@
+    let topology = m.value_of("TOPOLOGY").unwrap_or("standalone");
+    let (group, group_commentout) = match m.value_of("GROUP") {
+        Some(g) => (g, ""),
+        None => ("default", "#"),
+    };
+    let (config_secret_name, config_secret_name_commentout) =
+        match m.value_of("CONFIG_SECRET_NAME") {
+            Some(s) => (s, ""),
+            None => ("user-toml-secret", "#"),
+        };
+    let (ring_secret_name, ring_secret_name_commentout) = match m.value_of("RING_SECRET_NAME") {
+        Some(s) => (s, ""),
+        None => ("SECRET_NAME", "#"),
+    };
+    // clap_app!() ensures that we do have the mandatory args so unwrap() is fine here
+    let pkg_ident_str = m.value_of("PKG_IDENT").unwrap();
+    let pkg_ident = match PackageIdent::from_str(pkg_ident_str) {
+        Ok(pi) => pi,
+        Err(e) => return Err(format!("{}", e)),
+    };
+    let image = m.value_of("IMAGE").unwrap_or(pkg_ident_str);
+
+    let json = json!({
+        "metadata_name": pkg_ident.name,
+        "image": image,
+        "count": count,
+        "service_topology": topology,
+        "service_group_commentout": group_commentout,
+        "service_group": group,
+        "config_secret_name_commentout": config_secret_name_commentout,
+        "config_secret_name": config_secret_name,
+        "ring_secret_name_commentout": ring_secret_name_commentout,
+        "ring_secret_name": ring_secret_name,
+    });
+
+    let mut write: Box<Write> = match m.value_of("OUTPUT") {
+        Some(o) if o != "-" => {
+            match File::create(o) {
+                Ok(f) => Box::new(f),
+                Err(e) => return Err(format!("{}", e)),
+            }
+        }
+        _ => Box::new(io::stdout()),
+    };
+    if let Err(e) = Handlebars::new().template_renderw(MANIFESTFILE, &json, &mut write) {
+        Err(format!("{}", e))
+    } else {
+        Ok(())
+    }
+}
+
+fn cli<'a, 'b>() -> App<'a, 'b> {
+    let name: &str = &*PROGRAM_NAME;
+    clap_app!((name) =>
+        (about: "Creates a Kubernetes manifest for a Habitat package. Habitat \
+                 operator must be deployed within the Kubernetes cluster to \
+                 intercept the created objects.")
+        (version: VERSION)
+        (author: "\nAuthors: The Habitat Maintainers <humans@habitat.sh>\n\n")
+        (@arg IMAGE: --("image") +takes_value
+            "Image of the Habitat service exported as a Docker image")
+        (@arg OUTPUT: -o --("output") +takes_value
+            "Name of manifest file to create. Pass '-' for stdout (default: -)")
+        (@arg COUNT: --("count") +takes_value {valid_natural_number}
+            "Count is the number of desired instances")
+        (@arg TOPOLOGY: -t --("service-topology") +takes_value
+            "A topology describes the intended relationship between peers \
+             within a service group. Specify either standalone or leader \
+             topology (default: standalone)")
```

Comment:
We could validate this topology value

Reply:
Actually in the next PR i'm going to submit for also generating docker image, switches away from clap_app! and hence makes it possible to specify possible values for this option and I just made sure it does that. So if it's OK with you, i won't be changing this here.